### ISSUE TITLE
Having trouble installing a plugin/custom command through npm module. 

### ISSUE THREAD

**Issue by supun-hettigoda** [cypress-io/cypress#20633]
### Current behavior

I'm trying to migrate from cypress 6.8.0 to 9.5.0.
I have a docker file with contents

```
FROM cypress/included:9.5.0
RUN npm install cypress-iframe
ADD https://raw.githubusercontent.com/vishnubob/wait-for-it/master/wait-for-it.sh /wait-for-it.sh
RUN chmod +x /wait-for-it.sh
ENTRYPOINT ["/wait-for-it.sh"]
```

in my commands.js file I have 
`import 'cypress-iframe'`

now when I try to execute my tests I get an error
```
npm ERR! Tracker "idealTree" already exists

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-03-16T04_37_30_725Z-debug.log
```

This error goes away if I install the npm in a different location with 
WORKDIR /e2e  prior to npm install but then I get webpack errors 

```
./cypress/support/commands.js
Module not found: Error: Can't resolve 'cypress-iframe' in '/e2e/cypress/support'
resolve 'cypress-iframe' in '/e2e/cypress/support'
  Parsed request is a module
  No description file found
    Looked for and couldn't find the file at the following paths: or is not a directory
[/e2e/cypress/support/node_modules]
[/e2e/cypress/node_modules]
[/e2e/node_modules]
[/node_modules]
 @ ./cypress/support/commands.js 3:0-25
 @ ./cypress/support/index.js
 
    at handle (/root/.cache/Cypress/9.5.0/Cypress/resources/app/packages/server/node_modules/@cypress/webpack-preprocessor/dist/index.js:180:23)
    at finalCallback (/root/.cache/Cypress/9.5.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:257:39)
    at /root/.cache/Cypress/9.5.0/Cypress/resources/app/node_modules/webpack/lib/Compiler.js:306:14
```

I couldn't find any documentation to correctly install a plugin or a custom command via node package.
Any help would be much appreciated.

### Cypress Version

9.5.0

### Package Manager

npm

### Operating system

Linux

**Comment by emilyrohrbough** [comment]
@supun-hettigoda Could you provide more information on your project setup? I see you are installing the `cypress-iframe` module in the docker-image, however without more context, I am wondering why cypress-iframe` isn't listed as a dev-dependency in your project's package.json so you can run `npm install` or `yarn install`.

If you must keep it installed on your image, you likely need to install `cypress-iframe` globally so it's available within the project context.

**Comment by supun-hettigoda** [comment]
@emilyrohrbough  thanks for getting back to me on this.
Indeed some helpful tips for using `package.json` I will check those tips and see how it goes.  

I tried this but it didn't work. 
`If you must keep it installed on your image, you likely need to install cypress-iframe globally so it's available within the project context.`

I'm still not sure about the reason for first error and why it doesn't allow me to install npm packages in root directory.
But I managed to install the npm package in /e2e and get it imported using `import 'cypress-iframe'`  syntax.

The part I missed was the basic understanding of docker volume mounting.
**Lesson learned was :**  When you mount a host volume(ex: your spec folder) to a container volume (ex: in this case /e2e) remember that based on the docker container os (Linux) it will override all the contents in the docker container folder from host folder. So obviously node_module folder created in npm install will be gone when the container runs with a spec folder mounted.

**FIX** - was for me is to  when running the cypress container mount the spec folder into level up than which I installed npm package. so mounting it to  /e2e/app  solved it to me.
